src/vm/tests/simple_apply_eval.rs

```diff
@@ -283,6 +285,31 @@ fn test_options_errors() {
     }
 }
 
+#[test]
+fn test_some() {
+    let tests = [
+        "(eq? (some 1) (some 1))",
+        "(eq? none none)",
+        "(is-none? (some 1))",
+        "(eq? (some 1) none)",
+        "(eq? none (some 1))",
+        "(eq? (some 1) (some 2))",
```

Comment:
What would happen with `(eq? (some 1) (some 'true))`?  Would the typecheck fail?

Reply:
Yep -- I can add that to the error-case tests

Reply:
Resolved